File: PHP/5/example5.php
#!/usr/bin/env php

<?php

function display_names($name_list)
{
    echo "**************************************\n";
    foreach ($name_list as $name)
    {
        echo $name . "\n";
    }
    echo "\n";
}

function double_num($num)
{
    return $num * 2;
}

$names = ['Bob', 'Jack', 'Jill', 'Mike', 'John', 'Jim', 'Lisa'];
display_names($names);

$name_list = [];

foreach ($names as $name)
{
    array_push($name_list, $name);
}

display_names($name_list);

echo "Put Alfa at the front of the line ...\n";
array_unshift($name_list, "Alfa");
display_names($name_list);

echo "Remove Bob from the list ...\n";
unset($name_list[array_search("Bob", $name_list)]);
display_names($name_list);

echo "Remove the first and last elements ...\n";
$name_list = array_slice($name_list, 1, -1);
display_names($name_list);

echo "Clear the list ...\n";
$name_list = [];
display_names($name_list);

echo "Fill it with names again ...\n";
$name_list = $names;
display_names($name_list);

echo "The index of Mike is ...\n";
$index = array_search("Mike", $name_list);
echo $index . "\n\n";

echo "Get the count of 'Mike' in name_list ...\n";
$mikes = array_count_values($name_list);
print_r($mikes['Mike'] . "\n\n");

echo "Sort name_list by alphabetical order ...\n";
sort($name_list);
display_names($name_list);

echo "Reverse sort ...\n";
$reversed = array_reverse($name_list);
display_names($reversed);

echo "Slice the 2nd through 4th value ...\n";
$name_list = array_slice($name_list, 1, 3);
display_names($name_list);

echo "Inserting Charles at the 3rd position ...\n";
array_splice($name_list, 2, 0, "Charles");
display_names($name_list);

echo "Now removing Chares ...\n";
unset($name_list[2]);
display_names($name_list);

echo "Let's switch to numbers ...\n";
$num_list = [5, 7, 9, 2, 4, 6];
display_names($num_list);

echo "Now, let's double them ...\n";
$double_list = array_map("double_num", $num_list);
display_names($double_list);

echo "Let's do some sets and call it a day ...\n";
$fruits = ['banana', 'orange', 'apple', 'apple', 'grapes'];
$fruits = array_unique($fruits);
display_names($fruits);

File: Java/basic/src/main/java/org/prodigius/javabasic/example5/Example5.java
package org.prodigius.javabasic.example5;

import java.util.ArrayList;
import java.util.List;
import java.util.Collections;
import java.util.Set;
import java.util.HashSet;

public class Example5
{
    public static void displayNames(List<String> nameList)
    {
        System.out.println("************************************************");
        for (String name : nameList)
        {
            System.out.println(name);
        }
        System.out.println("");
    }

    public static void displayNums(List<Integer> numList)
    {
        System.out.println("************************************************");
        for (int num : numList)
        {
            System.out.println(num);
        }
        System.out.println("");
    }

    public static void main(String[] args)
    {
        List<String> names = new ArrayList<String>();
        names.add("Bob");
        names.add("Jack");
        names.add("Jill");
        names.add("Mike");
        names.add("John");
        names.add("Jim");
        names.add("Lisa");

        List<String> nameList = new ArrayList<String>();
        for (String name: names)
        {
            nameList.add(name);
        }
        displayNames(nameList);

        System.out.println("More names ...");
        List<String> moreNames = new ArrayList<String>();
        moreNames.add("Mark");
        moreNames.add("Matt");
        moreNames.add("Luke");
        moreNames.add("Juan");
        moreNames.add("Jose");

        for (String name : moreNames)
        {
            nameList.add(name);
        }

        displayNames(nameList);

        System.out.println("Put Alfa at the front of the line ...");
        nameList.add(0, "Alfa");
        displayNames(nameList);

        System.out.println("Remove Bob from the list ...");
        nameList.remove("Bob");
        displayNames(nameList);

        System.out.println("Remove the first and the last ...");
        nameList.remove(0);
        nameList.remove(nameList.size() - 1);
        displayNames(nameList);

        System.out.println("Clear the whole list ...");
        nameList.clear();
        displayNames(nameList);

        System.out.println("Fill with names again ...");
        for (String name : names)
        {
            nameList.add(name);
        }
        displayNames(nameList);

        System.out.println("Get the index of Mike ...");
        int mikeIndex = nameList.indexOf("Mike");
        System.out.println(mikeIndex);

        System.out.println("Get the count of 'Mike' occurences...");
        int mikeOccurences = Collections.frequency(nameList, "Mike");
        System.out.println(mikeOccurences);

        System.out.println("Sort by alphabetical order ...");
        Collections.sort(nameList);
        displayNames(nameList);

        System.out.println("Reverse the sort ...");
        Collections.sort(nameList, Collections.reverseOrder());
        displayNames(nameList);

        System.out.println("Insert Charles at the 3rd position ...");
        nameList.add(2, "Charles");
        displayNames(nameList);

        System.out.println("Now removing Charles ...");
        nameList.remove("Charles");
        displayNames(nameList);

        System.out.println("Let's switch to numbers ...");
        List<Integer> numList = new ArrayList<Integer>();
        numList.add(5);
        numList.add(7);
        numList.add(9);
        numList.add(2);
        numList.add(4);
        numList.add(6);
        displayNums(numList);

        System.out.println("Now, let's double them ...");
        List<Integer> doubleList = new ArrayList<Integer>();

        for (int num : numList)
        {
            doubleList.add(num * 2);
        }

        displayNums(doubleList);

        String[] fruits = {"banana", "orange", "apple", "apple", "grapes"};
        Set<String> fruitSet = new HashSet<String>();

        for (String fruit: fruits)
        {
            fruitSet.add(fruit);
        }

        ArrayList<String> fruitList = new ArrayList<String>();
        for (String fruit: fruitSet)
        {
            fruitList.add(fruit);
        }

        displayNames(fruitList);
    }
}